server/game/cards/01_SOR/units/AdmiralOzzelOverconfident.ts

```diff
@@ -0,0 +1,39 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { CardType, Location, Trait, RelativePlayer, WildcardCardType } from '../../../core/Constants';
+
+export default class AdmiralOzzelOverconfident extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '8117080217',
+            internalName: 'admiral-ozzel#overconfident'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addActionAbility({
+            title: 'Play an Imperial unit from your hand. It enters play ready',
+            cost: AbilityHelper.costs.exhaustSelf(),
+            immediateEffect: AbilityHelper.immediateEffects.sequential([
+                AbilityHelper.immediateEffects.selectCard({
+                    activePromptTitle: 'Play an Imperial unit from your hand. It enters play ready',
+                    cardTypeFilter: CardType.BasicUnit,
+                    // TODO: figure out how to make this assume that the played card must be from hand, unless specified otherwise
+                    locationFilter: Location.Hand,
```

Comment:
Actually, thinking on it further, probably that check should be in the play actions themselves since the `PlayType` makes it all the way to them. It was an oversight that they weren't already doing that check.

Can just put it in `PlayCardAction.meetsRequirements` since they inherit from that. I _think_ that should solve this issue and patch a few other holes as well.

Reply:
It looks like `PlayCardAction.meetsRequirements` already does that:
`if ( !ignoredRequirements.includes('location') && !context.player.isCardInPlayableLocation(context.source, this.playType)
        ) {
            return 'location';
        }`

Reply:
Yeah looks like you're right, commented in chat